adapter/slack.py

```diff
@@ -177,5 +183,38 @@ def bot_replied(self, event, parent_message):
         
         if chatbot is None:
             return {"text": self.MISSING_CHATBOT_ERROR}
-                
-        return asyncio.run(self.process_chatbot_request(chatbot, question, channel_id, thread_ts))
+        
+        history = self.get_chat_history(event)
+        return asyncio.run(self.process_chatbot_request(chatbot, question, channel_id, thread_ts, history))

```

Comment:
Okay I see some asynchronous operations going on. Will it handle race conditions and other errors gracefully?

Reply:
The operations primarily involve listing bots and asking questions to bots. Because of this, I think race conditions are less likely to be a major issue.

Reply:
I see